orbstation/code/customisation/preferences/languages.dm

```diff
@@ -0,0 +1,172 @@
+//LANGUAGE MENU BACKEND: FROM SKYRAT
+
+#define MAX_LANGUAGES_NORMAL 3
+#define MAX_LANGUAGES_LINGUIST 4
+
+//Language preference addition
+/datum/preferences
+	/// Associative list, keyed by language typepath, pointing to LANGUAGE_UNDERSTOOD, or LANGUAGE_SPOKEN, for whether we understand or speak the language
+	var/list/languages = list()
+
+/// This helper proc gets the current species language holder and does any post-processing that's required in one easy to track place.
+/// This proc should *always* be edited or used when modifying or getting the default languages of a player controlled, unrestricted species, to prevent any errant conflicts.
+/datum/preferences/proc/get_adjusted_language_holder()
+	var/datum/species/species = read_preference(/datum/preference/choiced/species)
+	species = new species()
+	var/datum/language_holder/language_holder = new species.species_language_holder()
+
+	return language_holder
+
+/datum/asset/spritesheet/languages
+	name = "languages"
+	early = TRUE
+	cross_round_cachable = TRUE
+
+/datum/asset/spritesheet/languages/create_spritesheets()
+	var/list/to_insert = list()
+
+	if(!GLOB.all_languages.len)
```

Comment:
Is this the same code from before? Put it in a proc so that when I ask you to make changes it updates both at once.